controllers/azurejson_machine_controller.go

```diff
@@ -91,13 +91,15 @@ func (f filterUnclonedMachinesPredicate) Generic(e event.GenericEvent) bool {
 	}
 
 	// when watching machines, we only care about machines users created one-off
-	// outside of machinedeployments/machinesets. if a machine is part of a machineset
+	// outside of machinedeployments/machinesets and using AzureMachineTemplates. if a machine is part of a machineset
 	// or machinedeployment, we already created a secret for the template. All machines
 	// in the machinedeployment will share that one secret.
 	_, isMachineSetNode := e.Meta.GetLabels()[clusterv1.MachineSetLabelName]
 	_, isMachineDeploymentNode := e.Meta.GetLabels()[clusterv1.MachineDeploymentLabelName]
```

Comment:
Unsure if we still need to check MachineSetLabel and MachineDeploymentLabel since machine deployments would also use templates and we'd get a cloned-from annotation.

Reply:
I think you're right. This controller is primarily meant to catch one off workers and control plane nodes not part of e.g. KCP. I haven't tested the logic but I think you can simplify it here. Whether a user creates MD or MS manually, they should get the cloned-from annotation.

Reply:
sounds good. will update

Reply:
I tested this and it seems to work

Reply:
updated to only check the cloned-from annotation.